fe1-web/store/reducers/SocialReducer.ts

```diff
@@ -0,0 +1,87 @@
+import { ChirpState } from 'model/objects/Chirp';
+import { createSelector, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Hash } from 'model/objects';
+import { getLaosState } from './LaoReducer';
+
+/**
+ * Stores all the Social Media related content
+ */
+
+// Stores all Social Media related information for a given LAO
+interface SocialReducerState {
+
+  // Stores all chirps for a given LAO
+  allChirps: ChirpState[];
+}
+
+// Root state for the Social Reducer
+interface SocialLaoReducerState {
+
+  // Associates a given LAO ID with the whole representation of its social media
+  byLaoId: Record<string, SocialReducerState>
+}
+
+const initialState: SocialLaoReducerState = {
+  byLaoId: {
+    myLaoId: {
+      allChirps: [],
+    },
+  },
+};
+
+const socialReducerPath = 'social';
+
+const socialSlice = createSlice({
+  name: socialReducerPath,
+  initialState,
+  reducers: {
+
+    // Add a chirp to the list of chirps
+    addChirp: {
+      prepare(laoId: Hash | string, chirp: ChirpState): any {
```

Comment:
why Hash | string?

Reply:
laoId.valueOf()

Reply:
Actually, it is directly converted to a value as you can see on the line just below this one. Then, it is just for convenience, so that we can call this function with an id as a Hash or as a string. In the codebase, the Hash is pretty much everywhere but in states, so we can omit any conversion when calling addChirp.